pkg/adapter/adapter.go

```diff
@@ -113,10 +115,13 @@ func (adp *LocalAdapter) Convert(ctx context.Context, source string) error {
 	if err = adp.content.NewRemoteCache(cacheRef); err != nil {
 		return err
 	}
+	adp.content.GcWaitGroup.Wait()
+	taskWaitGroup.Add(1)
 	if _, err = adp.cvt.Convert(ctx, source, target, cacheRef); err != nil {
```

Comment:
Should we ensure the wait group `Done()` when an error happens?

Reply:
Good point.